### PR TITLE
feature: Implement artifacts deletion

### LINKED ISSUE #744: Add support for deleting job artifacts
## Description of the problem, including code/CLI snippet
Add support for deleting a job's artifacts, via the `delete-artifacts` API call. 

## Expected Behavior
https://docs.gitlab.com/ee/api/jobs.html#delete-artifacts

## Actual Behavior
Not implemented

## Specifications
https://docs.gitlab.com/ee/api/jobs.html#delete-artifacts

### PR DESCRIPTION
**Pull request by gpocentek**
Closes #744

### CODE REVIEW

**Comment by max-wittig** [comment]
A line is still to long for the CI pipeline to pass.

**Comment by max-wittig** [comment]
@gpocentek Would you mind, rebasing or should I take over?

**Comment by gpocentek** [comment]
Sorry for the delay...

File: gitlab/v4/objects.py [hunk 1]
```diff
@@ -1584,6 +1584,21 @@ def keep_artifacts(self, **kwargs):
         path = "%s/%s/artifacts/keep" % (self.manager.path, self.get_id())
         self.manager.gitlab.http_post(path)
 
+    @cli.register_custom_action("ProjectJob")
+    @exc.on_http_error(exc.GitlabCreateError)
+    def delete_artifacts(self, **kwargs):
+        """Delete artifacts of a job?
```
Comment (max-wittig) [inline gitlab/v4/objects.py]: why the `?`

Comment (gpocentek) [reply gitlab/v4/objects.py]: typo!

**Comment by gpocentek** [comment]
This requires #791

**Comment by gpocentek** [comment]
I can't reproduce the tests failure locally...

**Comment by max-wittig** [comment]
I will merge this now. Tests are passing locally.

**Pull request merged**